functional/FunctionalTest.java

```diff
@@ -2299,660 +2307,368 @@ public static void threadedPutObject() throws Exception {
     }
   }
 
-  /** Test: copyObject(String bucketName, String objectName, String destBucketName). */
-  public static void copyObject_test1() throws Exception {
-    String methodName = "copyObject(String bucketName, String objectName, String destBucketName)";
+  public static void testCopyObject(
+      String testTags, ServerSideEncryption sse, CopyObjectArgs args, boolean negativeCase)
+      throws Exception {
+    String methodName = "copyObject()";
     if (!mintEnv) {
-      System.out.println("Test: " + methodName);
-    }
-
-    long startTime = System.currentTimeMillis();
-    try {
-      String objectName = getRandomName();
-      try (final InputStream is = new ContentInputStream(1 * KB)) {
-        client.putObject(bucketName, objectName, is, new PutObjectOptions(1 * KB, -1));
-      }
-
-      String destBucketName = getRandomName();
-      client.makeBucket(MakeBucketArgs.builder().bucket(destBucketName).build());
-      client.copyObject(destBucketName, objectName, null, null, bucketName, null, null, null);
-      client
-          .getObject(GetObjectArgs.builder().bucket(bucketName).object(objectName).build())
-          .close();
-
-      client.removeObject(RemoveObjectArgs.builder().bucket(bucketName).object(objectName).build());
-      client.removeObject(
-          RemoveObjectArgs.builder().bucket(destBucketName).object(objectName).build());
-      client.removeBucket(RemoveBucketArgs.builder().bucket(destBucketName).build());
-      mintSuccessLog(methodName, null, startTime);
-    } catch (Exception e) {
-      handleException(methodName, null, startTime, e);
+      System.out.println("Test: " + methodName + " " + testTags);
     }
-  }
 
-  /**
-   * Test: copyObject(String bucketName, String objectName, String destBucketName, CopyConditions
-   * copyConditions) with ETag to match.
-   */
-  public static void copyObject_test2() throws Exception {
-    if (!mintEnv) {
-      System.out.println(
-          "Test: copyObject(String bucketName, String objectName, String destBucketName,"
-              + "CopyConditions copyConditions) with Matching ETag (Negative Case)");
+    String srcObject = args.object();
+    if (args.srcObject() != null) {
+      srcObject = args.srcObject();
```

Comment:
Can be simplified using `Optional` class as follows:

`String srcObject = Optional.ofNullable(args.srcObject()).orElse(args.object());`

Reply:
Done